src/layers/tile-layer-utils/TileCache.js

```diff
@@ -126,4 +147,72 @@ export default class TileCache {
   _getTileId(x, y, z) {
     return `${z}-${x}-${y}`;
   }
+
+  _expandIndices(tileIndices){
+    var [minX, minY, maxY, maxX] = [Number.POSITIVE_INFINITY, Number.POSITIVE_INFINITY, Number.NEGATIVE_INFINITY, Number.NEGATIVE_INFINITY]
+    for (var tileIndex in tileIndices) {
+      var tile = tileIndices[tileIndex]
+      if(tile.x > maxX)
+        maxX = tile.x
+      if(tile.y > maxY)
+        maxY = tile.y
+      if(tile.y < minY)
+        minY = tile.y
+      if(tile.x < minX)
+        minX = tile.x
+    }
+    maxX = Math.min(maxX, Math.floor(MAX_HEIGHT / (256 * Math.pow(2, -1 * tileIndices[0].z))))
+    maxY = Math.min(maxY, Math.floor(MAX_WIDTH / (256 * Math.pow(2, -1 * tileIndices[0].z))))
+    var extraIndices = []
+    for (var tileIndex in tileIndices) {
+      var tile = tileIndices[tileIndex]
+      if (tile.x == maxX)
+        extraIndices.push({x: tile.x + 1, y: tile.y, z: tile.z})
+        if (tile.y == maxY)
+          extraIndices.push({x: tile.x + 1, y: tile.y + 1, z: tile.z})
+        else if (tile.y == minY)
+          extraIndices.push({x: tile.x + 1, y: tile.y - 1, z: tile.z})
+      else if (tile.x == minX)
+        extraIndices.push({x: tile.x - 1, y: tile.y, z: tile.z})
+        if (tile.y == maxY)
+          extraIndices.push({x: tile.x - 1, y: tile.y + 1, z: tile.z})
+        else if (tile.y == minY)
+          extraIndices.push({x: tile.x - 1, y: tile.y - 1, z: tile.z})
+      else if (tile.y == maxY)
+        extraIndices.push({x: tile.x, y: tile.y + 1, z: tile.z})
+      else if (tile.y == minY)
+        extraIndices.push({x: tile.x, y: tile.y - 1, z: tile.z})
```

Comment:
- Add `{}` around if-then clauses.
- DRY: I'd suggest just setting an `{x, y, z}` object inside the clauses, and push at the end.

Reply:
multiple of these statements can be hit in a single pass so we need to do some sort of list-pushing no matter what